pkg/nas/controllerserver.go

```diff
@@ -235,56 +369,169 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
-			return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: " + err.Error())
+	if nasVolArgs.VolumeAs == "filesystem" {
+		// protocalType
+		if nasVolArgs.ProtocalType, ok = volOptions[PROTOCAL_TYPE]; !ok {
+			nasVolArgs.ProtocalType = "NFS"
+		} else if nasVolArgs.ProtocalType != "NFS" {
+			return nil, fmt.Errorf("Required parameter [parameter.protocalType] must be [NFS]")
 		}
-		if !archiveBool {
-			if err := os.RemoveAll(deletePath); err != nil {
-				return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: " + err.Error())
+
+		// storageType
+		if nasVolArgs.StorageType, ok = volOptions[STORAGE_TYPE]; !ok {
+			nasVolArgs.StorageType = "Performance"
+		} else if nasVolArgs.StorageType != "Performance" && nasVolArgs.StorageType != "Capacity" {
+			return nil, fmt.Errorf("Required parameter [parameter.storageType] must be [Performance] or [Capacity]")
+		}
+
+		// zoneId
+		if nasVolArgs.ZoneId, ok = volOptions[ZONE_ID]; !ok {
+			nasVolArgs.ZoneId = GetMetaData(ZONEID_TAG)
+		}
+
+		// description
+		if nasVolArgs.Description, ok = volOptions[DESCRIPTION]; !ok {
+			nasVolArgs.Description = ""
+		}
+
+		// networkType
+		if nasVolArgs.NetworkType, ok = volOptions[NETWORK_TYPE]; !ok {
+			nasVolArgs.NetworkType = "vpc"
+		} else if nasVolArgs.NetworkType != "vpc" && nasVolArgs.NetworkType != "classic" {
+			return nil, fmt.Errorf("Required parameter [parameter.networkType] must be [vpc] or [classic]")
+		}
+
+		// vpcId
+		if nasVolArgs.VpcId, ok = volOptions[VPC_ID]; !ok {
+			if nasVolArgs.NetworkType == "vpc" {
+				return nil, fmt.Errorf("Required parameter [parameter.vpcId] must be set becasue [parameter.networkType] is [vpc]")
 			}
-			log.Infof("Delete Successful: Volume %s, Removed path %s", req.VolumeId, deletePath)
-			return &csi.DeleteVolumeResponse{}, nil
 		}
-	}
 
-	archivePath := filepath.Join(mountPoint, "archived-"+pvName+time.Now().Format(".2006-01-02-15:04:05"))
-	if err := os.Rename(deletePath, archivePath); err != nil {
-		log.Errorf("Delete Failed: Volume %s, archiving path %s to %s with error: %s", req.VolumeId, deletePath, archivePath, err.Error())
-		return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: ")
+		// vSwitchId
+		if nasVolArgs.VSwitchId, ok = volOptions[V_SWITCH_ID]; !ok {
+			if nasVolArgs.NetworkType == "vpc" {
+				return nil, fmt.Errorf("Required parameter [parameter.vSwitchId] must be set becasue [parameter.networkType] is [vpc]")
+			}
+		}
+
+		// accessGroupName
+		if nasVolArgs.AccessGroupName, ok = volOptions[ACCESSGROUP_NAME]; !ok {
+			nasVolArgs.AccessGroupName = "DEFAULT_VPC_GROUP_NAME"
+		}
+
+	} else if nasVolArgs.VolumeAs == "subpath" {
+		// server
+		if nasVolArgs.Server, ok = volOptions[SERVER]; !ok {
+			return nil, fmt.Errorf("Required parameter [parameter.server] must be set becasue [parameter.volumeAs] is [subpath]")
+		}
+
+		// mode
+		if nasVolArgs.Mode, ok = volOptions[MODE]; !ok {
```

Comment:
mode should not set default value. 
Mode empty means not to configure the file.